### ISSUE TITLE
ProgressMeter.next! prints as many bars as updates (on Jupyter notebook)

### ISSUE THREAD

**Issue by davidbp** [timholy/ProgressMeter.jl#71]
Hello,

When I use ``ProgressMeter.next!(progressbar; showvalues = [(:epoch, epoch), (:acc, acc)])`

I get the following behaviour

```
Progress:   5%|██                                       |  ETA: 0:00:13
  epoch:  1
Progress:  10%|████                                     |  ETA: 0:00:11
  epoch:  2
Progress:  15%|██████                                   |  ETA: 0:00:09
  epoch:  3
Progress:  20%|████████                                 |  ETA: 0:00:08
  epoch:  4
```

I was expecting a single bar that changed, not multiple of them.
I have tried with ``@showprogress``and it works as expected. This is the output:

```
fit!(percep, X_train, y_train, 20, 0.001)
Progress: 100%|█████████████████████████████████████████| Time: 0:00:07
```

For reproducible code here there is the notebook 
https://github.com/davidbp/jml/blob/master/MulticlassPerceptron/benchmarks/Perceptron_Benchmark.ipynb

**Comment by damiendr** [comment]
I have the same problem in IJulia and it seems to come from `showvalues`.

**Comment by timholy** [comment]
What version are you two running? If it's 0.5.1, can you try `Pkg.pin("ProgressMeter", v"0.5.0")`?

**Comment by timholy** [comment]
Oh, wait, this seems like a special case. Can you provide a complete minimum working example?

**Comment by damiendr** [comment]
The problem occurs with both 0.5.0 and 0.5.1 — I have IJulia 1.6.2.

Here is a working example:
```julia
using ProgressMeter
N = 100
p = Progress(100)
for i in 1:100
    update!(p, i; showvalues = [(:test, 1.0)])
    sleep(0.01)
end
```

Here as a notebook: https://gist.github.com/damiendr/4d932dfe60d8de17fbbe9a2d215edcfc#file-progressmeter-ipynb
Github renders it just as it appears on my machine except that I have the text in colors.

**Comment by juliohm** [comment]
I have the same issue on IJulia whenever I try to print some variable besides just showing the progress bar.

**Comment by marius311** [comment]
Dug into this since I was also hoping to have this work. Looks like it boils down to different behavior between notebook / REPL for:
![image](https://user-images.githubusercontent.com/1510968/38392355-83f75dd2-3927-11e8-8350-65aa2964e533.png)
and
![image](https://user-images.githubusercontent.com/1510968/38392379-99328d70-3927-11e8-9d76-64d8bfa4136c.png)

The same issue appears to be true of the Python version of this (`print("line1"); print("\033[F",end=''); print("line2")`) so its probably a general issue for Jupyter notebooks. 

I'm guessing Jupyter writes the output line-by-line and there's no way to go back once its up. But maybe there's a way to control the size of the output buffer? Anyone familiar with this?

**Comment by denizyuret** [comment]
Would it help solve the problem if we printed everything on one line?

**Comment by timholy** [comment]
If IJulia can't support the equivalent of terminal movement [escape codes](https://en.wikipedia.org/wiki/ANSI_escape_code) then one has to find a completely independent solution on IJulia. That will presumably involve learning about how Jupyter handles output to the browser.

**Comment by mboratko** [comment]
As pointed out, Python suffers from the same difficulty. In that case I was using [tqdm](https://github.com/tqdm/tqdm), and was able to easily switch to their tqdm_notebook functionality, which uses ipywidgets to display a much nicer progress bar that worked well within the notebook.

**Comment by VPetukhov** [comment]
I'm surprised that it's still an issue. I tried to replace `update!(p, i; showvalues = [(:test, 1.0)])` with

```julia
update!(p, i)
ProgressMeter.printover(p.output, "test: 1.0", p.color)
```
and indeed it's almost working. It shows the desired message for some time, but almost immediately it got overwritten with the default message. So, maybe, fixing https://github.com/timholy/ProgressMeter.jl/issues/33 could resolve this issue as well?

**Comment by oskarhenriksson** [comment]
Has this issue found a solution? 

When I try to run this in a Jupyter notebook in Visual Studio, several of the progress bars are printed:

```julia
p = Progress(5)
for i in 1:10
    sleep(0.3)
    next!(p, showvalues=[("Current iteration", i)])
end
```

<img width="856" alt="Image" src="https://github.com/user-attachments/assets/c49a55b1-a8d2-4131-bfe0-ded5ff9c4c79" />

As was pointed out above, the `showvalues` keyword seems to be crucial for the issue, since everything seems to work as expected if I just write `next!(p)`.

I'm using Julia 1.11.5 via the Julia VS Code extension v1.127.2 in Visual Studio Code 1.99.2 in Mac Sequoia 15.4.

**Comment by timholy** [comment]
It requires someone who uses IJulia and cares enough to decide to put in the effort to provide an implementation.

**Comment by MilesCranmer** [comment]
FYI this works for me just fine in Jupyter proper and Colab. I think the remaining issues are VSCode specific.

**Comment by 3f6a** [comment]
@MilesCranmer Yes, I think the relevant issue is https://github.com/microsoft/vscode/issues/118833. Unfortunately it seems there have been no recent updates.
